web/pgadmin/utils/__init__.py

```diff
@@ -805,9 +805,16 @@ def _does_user_exist(user, from_setup):
     This function will check user is exist or not. If exist then return
     """
     if isinstance(user, User):
-        user = user.email
-
-    new_user = User.query.filter_by(email=user).first()
+        if user.email:
```

Comment:
pgAdmin supports multiple authentication at a time, so it is possible to have same email ids and usernames for multiple users. 
So you have to check for username + Authentication source combination to get the valid user.

Reply:
@khushboovashi I have made changes :)

Reply:
I just didn't know what to do with email because the function accepts text or a user object. So for an object I check auth_source+username, but for text I check email.

Reply:
I am proposing to introduce auth_source as a user input while importing the server itself. So, you need change in setup.py itself. The fix you did does not resolve the issue.

Reply:
Sorry, I don't know pgadmin code at all. I tried to solve the problem, but I don't know if I can help more. Can you explain in more detail what we want to achieve? What should be changed in the file https://github.com/pgadmin-org/pgadmin4/blob/master/web/setup.py? What do you mean by user input?